<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>714</width>
    <height>573</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_3">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout">
          <item>
           <widget class="QTextBrowser" name="formula">
            <property name="font">
             <font>
              <family>Arial Rounded MT Bold</family>
              <pointsize>12</pointsize>
              <weight>50</weight>
              <bold>false</bold>
             </font>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Plain</enum>
            </property>
            <property name="html">
             <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Arial Rounded MT Bold'; font-size:12pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-family:'SimSun'; font-size:7.2pt;&quot;&gt;&lt;br /&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <item>
             <widget class="QLabel" name="sign">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>96</width>
                <height>0</height>
               </size>
              </property>
              <property name="text">
               <string/>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QTextBrowser" name="temp">
              <property name="minimumSize">
               <size>
                <width>0</width>
                <height>60</height>
               </size>
              </property>
              <property name="font">
               <font>
                <family>Arial Rounded MT Bold</family>
                <pointsize>14</pointsize>
                <weight>50</weight>
                <bold>false</bold>
               </font>
              </property>
              <property name="layoutDirection">
               <enum>Qt::LeftToRight</enum>
              </property>
              <property name="frameShadow">
               <enum>QFrame::Plain</enum>
              </property>
              <property name="html">
               <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Arial Rounded MT Bold'; font-size:14pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p align=&quot;right&quot; style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'SimSun'; font-size:7.2pt;&quot;&gt;0&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QGridLayout" name="button">
            <property name="sizeConstraint">
             <enum>QLayout::SetDefaultConstraint</enum>
            </property>
            <property name="horizontalSpacing">
             <number>7</number>
            </property>
            <property name="verticalSpacing">
             <number>0</number>
            </property>
            <item row="7" column="3">
             <widget class="Num_Button" name="btn_equl">
              <property name="font">
               <font>
                <family>Bahnschrift SemiBold SemiConden</family>
                <pointsize>20</pointsize>
                <weight>75</weight>
                <bold>true</bold>
               </font>
              </property>
              <property name="text">
               <string>=</string>
              </property>
             </widget>
            </item>
            <item row="4" column="0">
             <widget class="Num_Button" name="btn_7">
              <property name="font">
               <font>
                <family>Bahnschrift SemiBold SemiConden</family>
                <pointsize>20</pointsize>
                <weight>75</weight>
                <bold>true</bold>
               </font>
              </property>
              <property name="text">
               <string>7</string>
              </property>
             </widget>
            </item>
            <item row="3" column="0">
             <widget class="Num_Button" name="btn_left">
              <property name="font">
               <font>
                <family>Bahnschrift SemiBold SemiConden</family>
                <pointsize>20</pointsize>
                <weight>75</weight>
                <bold>true</bold>
               </font>
              </property>
              <property name="text">
               <string>(</string>
              </property>
             </widget>
            </item>
            <item row="3" column="1">
             <widget class="Num_Button" name="btn_right">
              <property name="font">
               <font>
                <family>Bahnschrift SemiBold SemiConden</family>
                <pointsize>20</pointsize>
                <weight>75</weight>
                <bold>true</bold>
               </font>
              </property>
              <property name="text">
               <string>)</string>
              </property>
             </widget>
            </item>
            <item row="6" column="0">
             <widget class="Num_Button" name="btn_1">
              <property name="font">
               <font>
                <family>Bahnschrift SemiBold SemiConden</family>
                <pointsize>20</pointsize>
                <weight>75</weight>
                <bold>true</bold>
               </font>
              </property>
              <property name="text">
               <string>1</string>
              </property>
             </widget>
            </item>
            <item row="3" column="3">
             <widget class="Num_Button" name="btn_div">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="font">
               <font>
                <family>Bahnschrift SemiBold SemiConden</family>
                <pointsize>20</pointsize>
                <weight>75</weight>
                <bold>true</bold>
               </font>
              </property>
              <property name="text">
               <string>/</string>
              </property>
             </widget>
            </item>
            <item row="6" column="3">
             <widget class="Num_Button" name="btn_add">
              <property name="font">
               <font>
                <family>Bahnschrift SemiBold SemiConden</family>
                <pointsize>20</pointsize>
                <weight>75</weight>
                <bold>true</bold>
               </font>
              </property>
              <property name="text">
               <string>+</string>
              </property>
             </widget>
            </item>
            <item row="5" column="3">
             <widget class="Num_Button" name="btn_sub">
              <property name="font">
               <font>
                <family>Bahnschrift SemiBold SemiConden</family>
                <pointsize>20</pointsize>
                <weight>75</weight>
                <bold>true</bold>
               </font>
              </property>
              <property name="text">
               <string>-</string>
              </property>
             </widget>
            </item>
            <item row="7" column="2">
             <widget class="Num_Button" name="btn_dot">
              <property name="font">
               <font>
                <family>Bahnschrift SemiBold SemiConden</family>
                <pointsize>20</pointsize>
                <weight>75</weight>
                <bold>true</bold>
               </font>
              </property>
              <property name="text">
               <string>.</string>
              </property>
             </widget>
            </item>
            <item row="4" column="1">
             <widget class="Num_Button" name="btn_8">
              <property name="font">
               <font>
                <family>Bahnschrift SemiBold SemiConden</family>
                <pointsize>20</pointsize>
                <weight>75</weight>
                <bold>true</bold>
               </font>
              </property>
              <property name="text">
               <string>8</string>
              </property>
             </widget>
            </item>
            <item row="3" column="2">
             <widget class="Num_Button" name="btn_clear">
              <property name="font">
               <font>
                <family>Bahnschrift SemiBold SemiConden</family>
                <pointsize>20</pointsize>
                <weight>75</weight>
                <bold>true</bold>
               </font>
              </property>
              <property name="text">
               <string>AC</string>
              </property>
             </widget>
            </item>
            <item row="7" column="0">
             <widget class="Num_Button" name="btn_bs">
              <property name="font">
               <font>
                <family>Bahnschrift SemiBold SemiConden</family>
                <pointsize>20</pointsize>
                <weight>75</weight>
                <bold>true</bold>
               </font>
              </property>
              <property name="text">
               <string>BS</string>
              </property>
             </widget>
            </item>
            <item row="6" column="1">
             <widget class="Num_Button" name="btn_2">
              <property name="font">
               <font>
                <family>Bahnschrift SemiBold SemiConden</family>
                <pointsize>20</pointsize>
                <weight>75</weight>
                <bold>true</bold>
               </font>
              </property>
              <property name="text">
               <string>2</string>
              </property>
             </widget>
            </item>
            <item row="6" column="2">
             <widget class="Num_Button" name="btn_3">
              <property name="font">
               <font>
                <family>Bahnschrift SemiBold SemiConden</family>
                <pointsize>20</pointsize>
                <weight>75</weight>
                <bold>true</bold>
               </font>
              </property>
              <property name="text">
               <string>3</string>
              </property>
             </widget>
            </item>
            <item row="5" column="1">
             <widget class="Num_Button" name="btn_5">
              <property name="font">
               <font>
                <family>Bahnschrift SemiBold SemiConden</family>
                <pointsize>20</pointsize>
                <weight>75</weight>
                <bold>true</bold>
               </font>
              </property>
              <property name="text">
               <string>5</string>
              </property>
             </widget>
            </item>
            <item row="5" column="2">
             <widget class="Num_Button" name="btn_6">
              <property name="font">
               <font>
                <family>Bahnschrift SemiBold SemiConden</family>
                <pointsize>20</pointsize>
                <weight>75</weight>
                <bold>true</bold>
               </font>
              </property>
              <property name="text">
               <string>6</string>
              </property>
             </widget>
            </item>
            <item row="5" column="0">
             <widget class="Num_Button" name="btn_4">
              <property name="font">
               <font>
                <family>Bahnschrift SemiBold SemiConden</family>
                <pointsize>20</pointsize>
                <weight>75</weight>
                <bold>true</bold>
               </font>
              </property>
              <property name="text">
               <string>4</string>
              </property>
             </widget>
            </item>
            <item row="4" column="3">
             <widget class="Num_Button" name="btn_mult">
              <property name="font">
               <font>
                <family>Bahnschrift SemiBold SemiConden</family>
                <pointsize>20</pointsize>
                <weight>75</weight>
                <bold>true</bold>
               </font>
              </property>
              <property name="text">
               <string>*</string>
              </property>
             </widget>
            </item>
            <item row="4" column="2">
             <widget class="Num_Button" name="btn_9">
              <property name="font">
               <font>
                <family>Bahnschrift SemiBold SemiConden</family>
                <pointsize>20</pointsize>
                <weight>75</weight>
                <bold>true</bold>
               </font>
              </property>
              <property name="text">
               <string>9</string>
              </property>
             </widget>
            </item>
            <item row="7" column="1">
             <widget class="Num_Button" name="btn_0">
              <property name="font">
               <font>
                <family>Bahnschrift SemiBold SemiConden</family>
                <pointsize>20</pointsize>
                <weight>75</weight>
                <bold>true</bold>
               </font>
              </property>
              <property name="text">
               <string>0</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="Num_Button" name="btn_factorial">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="font">
               <font>
                <family>Bahnschrift SemiBold SemiConden</family>
                <pointsize>20</pointsize>
                <weight>75</weight>
                <bold>true</bold>
               </font>
              </property>
              <property name="text">
               <string>!</string>
              </property>
             </widget>
            </item>
            <item row="0" column="0">
             <widget class="Num_Button" name="btn_pow">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="font">
               <font>
                <family>Bahnschrift SemiBold SemiConden</family>
                <pointsize>20</pointsize>
                <weight>75</weight>
                <bold>true</bold>
               </font>
              </property>
              <property name="text">
               <string>^</string>
              </property>
             </widget>
            </item>
            <item row="0" column="2">
             <widget class="Num_Button" name="btn_pi">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="font">
               <font>
                <family>Bahnschrift SemiBold SemiConden</family>
                <pointsize>20</pointsize>
                <weight>75</weight>
                <bold>true</bold>
               </font>
              </property>
              <property name="text">
               <string>π</string>
              </property>
             </widget>
            </item>
            <item row="0" column="3">
             <widget class="Num_Button" name="btn_e">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="font">
               <font>
                <family>Bahnschrift SemiBold SemiConden</family>
                <pointsize>20</pointsize>
                <weight>75</weight>
                <bold>true</bold>
               </font>
              </property>
              <property name="text">
               <string>e</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QTextBrowser" name="history">
      <property name="font">
       <font>
        <family>Arial</family>
        <pointsize>18</pointsize>
       </font>
      </property>
      <property name="html">
       <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Arial'; font-size:18pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p align=&quot;right&quot; style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-family:'SimSun'; font-size:7.2pt;&quot;&gt;&lt;br /&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>714</width>
     <height>26</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>开始</string>
    </property>
    <addaction name="actionexit"/>
    <addaction name="actionclear_history"/>
   </widget>
   <addaction name="menu"/>
  </widget>
  <action name="actionexit">
   <property name="text">
    <string>退出</string>
   </property>
  </action>
  <action name="actionclear_history">
   <property name="text">
    <string>清除历史</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Num_Button</class>
   <extends>QPushButton</extends>
   <header location="global">num_button.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
